src/udp.c

```diff
@@ -233,11 +233,48 @@ udp_bind ( int subsystem, const char *name,
       }
 #endif
 
-      if (setsockopt(fd, udp_get_solip(), IP_ADD_MEMBERSHIP, &m, sizeof(m))) {
-        inet_ntop(AF_INET, &m.imr_multiaddr, buf, sizeof(buf));
-        tvhwarn(subsystem, "%s - cannot join %s [%s]",
-                name, buf, strerror(errno));
+#if defined(PLATFORM_LINUX)
```

Comment:
I would move the code outside PLATFORM_LINUX - or ms.imr_sourceaddr does not exist on BSD / Darwin ?